src/umzug.ts

```diff
@@ -379,6 +421,127 @@ export class Umzug<Ctx> extends emittery.Typed<
+		allowConfusingOrdering?: boolean;
+		skipVerify?: boolean;
+	}): Promise<void> {
+		const isoDate = new Date().toISOString();
+		const prefixes = {
+			TIMESTAMP: isoDate.replace(/\.\d{3}Z$/, '').replace(/\W/g, '.'),
+			DATE: isoDate.split('T')[0].replace(/\W/g, '.'),
+			NONE: '',
+		};
+		const prefixType = options.prefix ?? 'TIMESTAMP';
+		const fileBasename = [prefixes[prefixType], options.name].filter(Boolean).join('.');
+
+		const allowedExtensions = options.allowExtension
+			? [options.allowExtension]
+			: ['.js', '.cjs', '.mjs', '.ts', '.sql'];
+
+		const existing = await this.migrations();
+		const last = existing[existing.length - 1];
+
+		const confusinglyOrdered = existing.find(e => e.path && path.basename(e.path) > fileBasename);
+		if (confusinglyOrdered && !options.allowConfusingOrdering) {
+			throw new Error(
+				`Can't create ${fileBasename}, since it's unclear if it should run before or after existing migration ${confusinglyOrdered.name}. Use allowConfusingOrdering to bypass this error.`
+			);
+		}
+
+		const folder = options.folder || this.options.create?.folder || (last?.path && path.dirname(last.path));
+
+		if (!folder) {
+			throw new Error(`Couldn't infer a directory to generate migration file in. Pass folder explicitly`);
+		}
+
+		const filepath = path.join(folder, fileBasename);
+
+		const template = this.options.create?.template ?? Umzug.defaultCreationTemplate;
+
+		const toWrite = template(filepath);
+		if (toWrite.length === 0) {
+			toWrite.push([filepath, '']);
+		}
+
+		toWrite.forEach(pair => {
+			if (!Array.isArray(pair) || pair.length !== 2) {
+				throw new Error(
+					`Expected [filepath, content] pair. Check that the file template function returns an array of pairs.`
+				);
+			}
+
+			const ext = path.extname(pair[0]);
+			if (!allowedExtensions.includes(ext)) {
+				const allowStr = allowedExtensions.join(', ');
+				const message = `Extension ${ext} not allowed. Allowed extensions are ${allowStr}. See help for allowExtension to avoid this error.`;
+				throw new Error(message);
+			}
+
+			fs.mkdirSync(path.dirname(pair[0]), { recursive: true });
+			fs.writeFileSync(pair[0], pair[1]);
+			this.logging({ event: 'created', path: pair[0] });
+		});
+
+		if (!options.skipVerify) {
+			const pending = await this.pending();
+			if (!pending.some(p => p.path && path.resolve(p.path) === path.resolve(filepath))) {
+				throw new Error(
+					`Expected ${filepath} to be a pending migration but it wasn't! You should investigate this. Use skipVerify to bypass this error.`
+				);
+			}
+		}
+	}
+
+	private static defaultCreationTemplate(filepath: string): Array<[string, string]> {
+		const dedent = (content: string) =>
+			content
+				.split('\n')
+				.map(line => line.trim())
+				.join('\n')
+				.trimStart();
+
+		const ext = path.extname(filepath);
+		if (ext === '.js' || ext === '.cjs') {
+			const content = dedent(`
+				exports.up = params => {};
+				exports.down = params => {};
```

Comment:
What do you think of being explicit to people that promises work?

```suggestion
				exports.up = async params => {};
				exports.down = async params => {};
```